src/katsdpcontroller/generator.py

```diff
@@ -208,6 +208,59 @@ def aggregate_compute(self) -> Tuple[Sensor.Status, int]:
         return (status, self._total)
 
 
+class SyncSensor(SimpleAggregateSensor[bool]):
+    """Aggregate which tracks whether its children are synchronised.
+
+    In this case,
+    - a 'synchronised' child has a reading of (True, NOMINAL), and
+    - an 'unsynchronised' child has a reading of (False, ERROR).
+    """
+
+    def __init__(
+        self, target: SensorSet, sensor_type: Type[bool],
```

Comment:
I'd even suggest removing the type from this class, and just hard-coding it to `bool` when you call `super().__init__()` lower down.

It's unlikely that the user would need to change this.

I'm tempted to suggest renaming this to something like `AllSensor` or something to do with its boolean-ness, because it could well be reused for other things that aren't sync statuses.

But looking at the ICD, there don't seem to be other potential candidates so perhaps it's best not to worry about that at this stage.

Reply:
@bmerry permission to update the corresponding `sensor_type` arg in `SumSensor` to `int` alongside this?

Reply:
Yes, definitely worth making things consistent. I think I'd originally thought SumSensor should be generic (so that we could accumulate `float` as well), but floating-point rounding errors don't play nicely with the implementation (continually adding things in then subtracting them out later).